### PR TITLE
Updated contracts

### PR DESCRIPTION
**Pull request by tulustul**
Added LiquidityAdditionClass
Added support for contracts versions
Added dynamic fee for stableswaps

### CODE REVIEW

File: pactsdk/pool_state.py [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
 @dataclass
 class AppInternalState:
-    """The one to one representation of pool's global state.
+    """The one to one representation of pool's global state."""
 
-    The optional properties are used only by stableswaps and should not be relevant to the users.
-    """
+    # Name and version may be missing in older contracts.
+    CONTRACT_NAME: Optional[Literal["PACT AMM", "[SI] PACT AMM"]]
+    VERSION: Optional[int]
```
Comment (Sernikjamnika) [inline pactsdk/pool_state.py:21]: how about moving it to line 33 to have all "may be missing" togehter?

File: pactsdk/stableswap_calculator.py [hunk 2]
```diff
@@ -30,36 +31,271 @@ class StableswapParams:
     precision: int
 
 
+def get_tx_fee(swap_invariant_iterations: int, extra_margin: int) -> int:
+    """To calculate the pool invariant, a Newton-Raphson method is used in both - the SDK and the smart contract.
+
+    Algorand has a limit of the number of operations available in a single app call. To increase the limit, an additional empty inner transaction have to be created. Each extra tx increases tx fee. This functions calculates the fee needed for a swap transaction.
```
[the remark this answers is not in the document]
Comment (Sernikjamnika) [reply pactsdk/stableswap_calculator.py:37]: Not only swap but add liq as well. If I didn't cought that in JS-SDK it's worth changing there as well.

File: pactsdk/stableswap_calculator.py [hunk 3]
```diff
@@ -30,36 +31,271 @@ class StableswapParams:
...
+    future_a: int,
+    future_a_time: int,
+) -> int:
+    """Calculates the amplifier at specified timestamp using linear interpolation."""
+    dt = future_a_time - initial_a_time
+    dv = future_a - initial_a
+    if not dt or not dv:
+        return future_a
+
+    dv_per_ms = dv / dt
```
Comment (Sernikjamnika) [inline pactsdk/stableswap_calculator.py:263]: rather cosmetic but dv_per_s

File: pactsdk/constant_product_calculator.py [hunk 4]
```diff
@@ -2,13 +2,46 @@
...
+def get_swap_amount_deposited(
+    liq_a: int,
+    liq_b: int,
+    gross_amount_received: int,
+) -> int:
+    deposited = liq_a * gross_amount_received / (liq_b - (gross_amount_received))
+    return math.ceil(deposited)
+
+
+def get_constant_product_minted_liquidity_tokens(
```
[the remark this answers is not in the document]
Comment (tulustul) [reply pactsdk/constant_product_calculator.py:32]: discussed verbally

**Review by Sernikjamnika — APPROVED** [review]

**Pull request merged**